src/System.Security.Cryptography.Encoding/tests/DerSequenceReaderTests.cs

```diff
@@ -78,5 +79,20 @@ public static void ReadOids()
             // And... done.
             Assert.False(reader.HasData);
         }
+
+
+        [Theory]
+        [InlineData("170d3135303430313030303030305a", 2015, 4, 1, 0, 0, 0)] // Tests encoding for midnight
+        [InlineData("170d3439313233313131353935395a", 2049, 12, 31, 11, 59, 59)]
+        [InlineData("170d3531303632383030303330305a", 1951, 06, 28, 0, 3, 0)]
+        public static void ReadUtcTime_Midnight(string hexString, int year, int month, int day, int hour, int minute, int second)
+        {
+            byte[] payload = hexString.HexToByteArray();
+            DateTime representedTime = new DateTime(year, month, day, hour, minute, second);
```

Comment:
Oh, I know that; I'm merely pointing out that amid a sea of caring about the DateTimeKind of dates you let an unspecified creep in.  Since you actually expect that it will match that DateTime (UTC) I'd specify the DateTimeKind; or give a comment as to why this is (or, I suppose, "emphatically isn't") special.